src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  actions.forEach((action) => {
+    const previousState =
+      states.length > 0 ? states[states.length - 1] : { ...state };
```

Comment:
Instead of checking the length of states each time inside the forEach loop, you can initialize the newState with the original state outside of the loop and then simply clone the newState at the end of each iteration. This will avoid unnecessary repeated checks and make the code cleaner.